src/data/init-moves.ts

```diff
@@ -1946,6 +1950,10 @@ export function initMoves() {
       HighCritAttr,
     ),
     new SelfStatusMove(MoveId.LUNAR_DANCE, ElementalType.PSYCHIC, -1, 10, -1, 0, 4)
+      .condition(
+        (user, _target, _move) =>
+          user.getParty().filter((p) => !p.isFainted()).length > (globalScene.currentBattle.double ? 1 : 0),
```

Comment:
```suggestion
          user.getParty().filter((p) => !p.isFainted() && p.isAllowedInChallenge()).length > (globalScene.currentBattle.double ? 1 : 0),
```

(same about the comment)